### PR TITLE
Constantinople

### PR DESCRIPTION
**Pull request by dwightguth**
EXTCODEHASH hasn't been fully tested yet because those tests haven't been merged upstream yet, but I will make a followup PR updating the tests submodule and fixing any bugs when it does. The rest have been fully tested and seem to pass, with the exception of two tests that expend more memory than we have on the machine, and two tests that seem to have incorrectly filled result information (which I have filed as a bug on the test repo)

### CODE REVIEW

**Comment by dwightguth** [comment]
@daejunpark I need your help figuring out how to update the specifications so that they will be correct for constantinople.

File: evm.md [hunk 1]
```diff
@@ -2105,16 +2183,42 @@ There are several helpers for calculating gas (most of them also specified in th
...
+             #else
+                 0
+             #fi +Int
+             #if CURR =/=Int NEW andBool ORIG ==Int NEW #then 
+                 #if ORIG ==Int 0 #then Gsstoreset < SCHED > #else Gsstorereset < SCHED > #fi -Int Gsload < SCHED >
+             #else
+                 0
+             #fi
+         #fi
+      requires Ghasdirtysstore << SCHED >>
```
Comment (ehildenb) [inline evm.md]: Several whitespace errors introduced here.

File: data.md [hunk 2]
```diff
@@ -320,11 +324,18 @@ Bitwise logical operators are lifted from the integer versions.
                  | Int "|Word"   Int [function]
                  | Int "&Word"   Int [function]
                  | Int "xorWord" Int [function]
+                 | Int "<<Word"  Int [function]
+                 | Int ">>Word"  Int [function]
+                 | Int ">>sWord" Int [function]
```
Comment (ehildenb) [inline data.md]: whitespace errors

**Comment by ehildenb** [comment]
Jenkins: test mantis

**Comment by ehildenb** [comment]
Jenkins: test this please

**Comment by ehildenb** [comment]
Looks good, not sure why tests are failing.

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test mantis

**Comment by dwightguth** [comment]
Jenkins: test mantis

**Comment by dwightguth** [comment]
Jenkins: test mantis

**Comment by dwightguth** [comment]
For context, the mantis build failed because of a bug in an updated package on rvwork2 and the standalone build failed because of an intermittent bug involving serialization that we may need to investigate further at some point, but I fixed the former and the latter doesn't have anything to do with this PR, so I just reran everything. It's gotten past the points where it crashed before, so I think it's going to succeed now. I will merge when the tests pass.

**Pull request merged**